src/Role/Repository/RoleRankRepository.php

```diff
@@ -33,60 +35,49 @@ class RoleRankRepository extends BaseRepository {
     /**
      * @param int $rankId
      *
-     * @return array|null
-     * @throws QueryException
+     * @return int|null
      */
-    public function getRankRoleIds(int $rankId) : ?array {
+    public function getRoleId(int $rankId) : ?int {
+
         $searchCriteria = $this->getDataObjectManager()
-        ->select('role_id')
         ->where('rank_id', $rankId);
 
-        return $this->getList($searchCriteria)->get('role_id');
+        /** @var RoleRank $roleRank */
+        $roleRank = $this->getOneBy($searchCriteria, true);
+
+        if(!$roleRank) {
+            return null;
+        }
+
+        return $roleRank->getRoleId();
     }
 
     /**
      * @param int $roleId
-     * @param int $rankId
      *
-     * @return RoleRank|null
-     * @throws NoSuchEntityException
+     * @return array|null
+     * @throws QueryException
      */
-    public function getRankAssignedRole(int $roleId, int $rankId): ?RoleRank {
-        $searchCriteria = $this->getDataObjectManager()->where([
-            RoleRankInterface::COLUMN_ROLE_ID => $roleId,
-            RoleRankInterface::COLUMN_RANK_ID => $rankId
-        ]);
-
-        return $this->getOneBy($searchCriteria, true, false);
+    public function getRoleRankIds(int $roleId) : ?array {
+        $searchCriteria = $this->getDataObjectManager()
+        ->select('rank_id')
+        ->where('role_id', $roleId);
+        //TODO this must be only for one
+        return $this->getList($searchCriteria)->get('rank_id');
     }
 
     /**
-     * @param array $allRankRoleIds
+     * @param int $roleId
+     * @param int $rankId
      *
-     * @return array|null
+     * @return RoleRank|null
+     * @throws NoSuchEntityException
      */
-    public function getRankPermissions(array $allRankRoleIds): ?array
-    {
+    public function getRankAssignedRole(int $roleId, int $rankId): ?RoleRank {
         $searchCriteria = $this->getDataObjectManager()
-            ->select([
-                'ares_roles_permission.id',
-                'ares_roles_permission.role_id',
-                'ares_roles_permission.permission_id'
-            ])->from('ares_roles_permission')
-            ->leftJoin(
-                'ares_permissions',
-                'ares_permissions.id',
-                '=',
-                'ares_roles_permission.permission_id'
-            )->whereIn(
-                'ares_roles_permission.role_id',
-                $allRankRoleIds
-            )->select('ares_permissions.name');
-
-        return array_values(
-            array_unique(
-                $this->getList($searchCriteria)->get('name') ?? []
-            )
-        );
+        ->where(RoleRankInterface::COLUMN_ROLE_ID, '=', $roleId)
+        ->orWhere(RoleRankInterface::COLUMN_RANK_ID, '=', $rankId);
+        //TODO I dont like this...
+        return $this->getOneBy($searchCriteria, true, false);
```

Comment:
Do you want to change that todo ?

Reply:
Honestly I don't remember why I didn't like that, maybe some Cache Issues that's why I don't like it I can just move that TODO onto DocBlock just as Dome said